resources/cluster-essentials/charts/delete-helm-release/templates/xip-patch.yaml

```diff
@@ -0,0 +1,38 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ template "delete-helm-release.fullname" . }}-xip-patch-job
+  namespace: kyma-installer
+  annotations:
+    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
+    helm.sh/hook: "post-upgrade"
+    helm.sh/hook-weight: "10"
+  labels:
+    chart: {{ template "delete-helm-release.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  activeDeadlineSeconds: 1000
+  backoffLimit: 3
```

Comment:
Since we have retries implemented in the script, I don't think we should retry the whole job. Either make the job atomic (either it succeeds as a whole, or fails and is restarted as a whole) and use backoffLimit, or keep the partial retries in the job and disable backoff

Reply:
If you don't mind, I would leave some backOffLimit - perhaps reduced to 2.
First, there are some [bugs](https://github.com/kubernetes/kubernetes/issues/67848) related to this setting with a value of zero.
Second, local re-tries in the script are meant to handle temporary networking failures.
There are many more reasons for "failures" in k8s world - e.g: Pod being moved between nodes or a Node failure, as explained [here](https://kubernetes.io/docs/concepts/workloads/controllers/job/#handling-pod-and-container-failures).
That's why I'd prefer to re-run this job at least once in case of a failure, especially on production.
I will decrease activeDeadlineSeconds to 60 to ensure it will not "block" the upgrade process.